rules/Php80/Rector/Catch_/RemoveUnusedVariableInCatchRector.php

```diff
@@ -86,4 +103,28 @@ public function provideMinPhpVersion(): int
     {
         return PhpVersionFeature::NON_CAPTURING_CATCH;
     }
+
+    private function isUsedInNextStmt(StmtsAwareInterface $stmtsAware, int $jumpToKey, string $variableName): bool
+    {
+        if ($stmtsAware->stmts === null) {
+            return false;
+        }
+
+        $totalKeys = array_key_last($stmtsAware->stmts);
+        for ($key = $jumpToKey; $key <= $totalKeys; ++$key) {
+            if (! isset($stmtsAware->stmts[$key])) {
+                continue;
+            }
+
+            $isVariableUsed = (bool) $this->betterNodeFinder->findVariableOfName(
+                $stmtsAware->stmts[$key],
+                $variableName
+            );
+            if ($isVariableUsed) {
+                return true;
+            }
+        }
+
+        return false;
+    }
```

Comment:
This looks like useful re-usable service. Well done :+1: 

Could you extact it to we can use it in another places?

Reply:
Ok, I moved to `StmtsManipulator` service https://github.com/rectorphp/rector-src/pull/4459/commits/5685c62d3ea3da15f162c100bd0a8d9a7a328135